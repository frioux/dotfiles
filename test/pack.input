#!/bin/bash

dir="$PWD/test/,pack"
makefile="$dir/Makefile"

mkdir -p "$dir"
cat >"$makefile" <<END
include $PWD/Makefile

.PHONY: test-pack
test-pack:
	@mv \$(archive_name) '$dir'
	@unzip -Z1 '$dir/\$(archive_name)' \
	 | sed 's!^\$(archive_basename)/!!' \
	 | grep -v '/\$\$' \
	 | sort \
	 >'$dir/pack'
	@for i in \$(targets_all_archived); do echo "\$\$i"; done \
         | sort \
         >'$dir/orig'
	@diff '$dir/orig' '$dir/pack'; true
END

make --always-make -f "$makefile" pack >/dev/null
make --always-make -f "$makefile" test-pack

rm -rf "$dir"

# __END__
