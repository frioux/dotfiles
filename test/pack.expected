GENERATE .mduem/cache/Makefile.variables
0a1
> 
